src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3744,6 +3744,14 @@ theorem toInt_abs_eq_natAbs_of_ne_intMin {x : BitVec w} (hx : x ≠ intMin w) :
     x.abs.toInt = x.toInt.natAbs := by
   simp [toInt_abs_eq_natAbs, hx]
 
+theorem toFin_abs {x : BitVec w} :
+    x.abs.toFin = Fin.ofNat' (2 ^ w) x.abs.toNat  := by
```

Comment:
This could, alternatively be: `= if x.msb then Fin.ofNat' (2 ^ w) (2^w - x.toNat) else x.toFin`.

I don't know the conventions very well. Did you think about this alternative, and do you know which one is preferable?

Reply:
The full statement and proof is:

```lean
theorem toFin_abs' {x : BitVec w} :
    x.abs.toFin = if x.msb then Fin.ofNat' (2 ^ w) (2^w - x.toFin) else x.toFin := by
  rw [BitVec.abs]
  by_cases hx : x.msb
  · simp [hx]
  · simp [hx, ← BitVec.toFin_ofNat]
```

Reply:
I had not considered that rhs, but it seems more consistent with the `toNat` and `toInt` definitions, so I'll switch to that, thanks!